components/dropdown_input.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, CSSProperties} from 'react';
+import ReactSelect, {Props as SelectProps, ActionMeta, components} from 'react-select';
+import classNames from 'classnames';
+
+import './dropdown_input.scss';
+
+// TODO: This component needs work, should not be used outside of AddressInfo until this comment is removed.
+
+type ValueType = {
+    label: string;
+    value: string;
+}
+
+type Props<T> = Omit<SelectProps<T>, 'onChange'> & {
+    value?: T;
+    legend?: string;
+    error?: string;
+    onChange: (value: T, action: ActionMeta<T>) => void;
+};
+
+const baseStyles = {
+    input: (provided: CSSProperties) => ({
+        ...provided,
+        color: 'var(--center-channel-color)',
+    }),
+    control: (provided: CSSProperties) => ({
+        ...provided,
+        border: 'none',
+        boxShadow: 'none',
+        padding: '0 2px',
+    }),
+    indicatorSeparator: (provided: CSSProperties) => ({
+        ...provided,
+        display: 'none',
+    }),
+};
+
+const IndicatorsContainer = (props: any) => {
+    return (
+        <div className='DropdownInput__indicatorsContainer'>
+            <components.IndicatorsContainer {...props}>
+                <i className='icon icon-chevron-down'/>
+            </components.IndicatorsContainer>
+        </div>
+    );
+};
+
+const Option = (props: any) => {
+    return (
+        <div
+            className={classNames('DropdownInput__option', {
+                selected: props.isSelected,
+                focused: props.isFocused,
+            })}
+        >
+            <components.Option {...props}/>
+        </div>
+    );
+};
+
+const renderError = (error?: string) => {
+    if (!error) {
+        return null;
+    }
+
+    return (
+        <div className='Input___error'>
+            <i className='icon icon-alert-outline'/>
+            <span>{error}</span>
+        </div>
+    );
+};
+
+const DropdownInput = <T extends ValueType>(props: Props<T>) => {
+    const {value, placeholder, className, addon, name, textPrefix, legend, onChange, styles, options, error, ...otherProps} = props;
+
+    const [focused, setFocused] = useState(false);
+
+    const onInputFocus = (event: React.FocusEvent<HTMLElement>) => {
+        const {onFocus} = props;
+
+        setFocused(true);
+
+        if (onFocus) {
+            onFocus(event);
+        }
+    };
+
+    const onInputBlur = (event: React.FocusEvent<HTMLElement>) => {
+        const {onBlur} = props;
+
+        setFocused(false);
+
+        if (onBlur) {
+            onBlur(event);
+        }
+    };
+
+    let inputClass = className ? `Input ${className}` : 'Input';
```

Comment:
Would it be better to use pkg Classnames here? might help inline these as opposed to using the ternary operator.

Reply:
Yeah, this was pulled from CWS where I don't think we're using this. Should be a quick refactor.